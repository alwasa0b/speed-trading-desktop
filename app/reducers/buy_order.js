import {
  UPDATE_QUANTITY,
  UPDATE_SELL_ORDER_TYPE,
  UPDATE_BUY_ORDER_TYPE,
  UPDATE_BUY_PRICE,
  UPDATE_SELL_PRICE
} from "../constants/buy";

export default (
  state = {
    quantity: "",
    sell_order_type: "none",
    sell_price: "",
    buy_order_type: "bid",
    buy_price: ""
  },
  action
) => {
  switch (action.type) {
    case UPDATE_QUANTITY:
      return { ...state, quantity: action.quantity };
    case UPDATE_SELL_ORDER_TYPE:
      return { ...state, sell_order_type: action.sell_order_type };
    case UPDATE_BUY_ORDER_TYPE:
      return { ...state, buy_order_type: action.buy_order_type };
    case UPDATE_BUY_PRICE:
      return { ...state, buy_price: action.buy_price };
    case UPDATE_SELL_PRICE:
      return { ...state, sell_price: action.sell_price };
    default:
      return state;
  }
}
