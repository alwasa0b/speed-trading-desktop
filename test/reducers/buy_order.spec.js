import buy_order from "../../app/reducers/buy_order";
import {
  update_quantity,
  update_quantity_type,
  update_sell_order_type,
  update_buy_order_type,
  update_buy_price,
  update_sell_price
} from "../../app/actions/buy";

describe("reducers", () => {
  describe("buy_order", () => {
    it("should handle initial state", () => {
      expect(buy_order(undefined, {})).toMatchSnapshot();
    });

    it("should handle update_quantity", () => {
      expect(
        buy_order(undefined, update_quantity({ quantity: 1 }))
      ).toMatchSnapshot();
    });

    it("should handle update_quantity_type", () => {
      expect(
        buy_order(
          undefined,
          update_quantity_type({ quantity_type: "quantity_test_type" })
        )
      ).toMatchSnapshot();
    });

    it("should handle update_sell_order_type", () => {
      expect(
        buy_order(
          undefined,
          update_sell_order_type({ sell_order_type: "sell_order_test_type" })
        )
      ).toMatchSnapshot();
    });

    it("should handle update_buy_order_type", () => {
      expect(
        buy_order(
          undefined,
          update_buy_order_type({ buy_order_type: "buy_order_test_type" })
        )
      ).toMatchSnapshot();
    });

    it("should handle update_buy_price", () => {
      expect(
        buy_order(undefined, update_buy_price({ buy_price: 2 }))
      ).toMatchSnapshot();
    });

    it("should handle update_sell_price", () => {
      expect(
        buy_order(undefined, update_sell_price({ sell_price: 3 }))
      ).toMatchSnapshot();
    });
  });
});
