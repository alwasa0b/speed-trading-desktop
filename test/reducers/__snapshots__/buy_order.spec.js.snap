// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers buy_order should handle initial state 1`] = `
Object {
  "buy_order_type": "bid",
  "buy_price": "",
  "quantity": "",
  "quantity_type": "count",
  "sell_order_type": "none",
  "sell_price": "",
}
`;

exports[`reducers buy_order should handle update_buy_order_type 1`] = `
Object {
  "buy_order_type": "buy_order_test_type",
  "buy_price": "",
  "quantity": "",
  "quantity_type": "count",
  "sell_order_type": "none",
  "sell_price": "",
}
`;

exports[`reducers buy_order should handle update_buy_price 1`] = `
Object {
  "buy_order_type": "bid",
  "buy_price": 2,
  "quantity": "",
  "quantity_type": "count",
  "sell_order_type": "none",
  "sell_price": "",
}
`;

exports[`reducers buy_order should handle update_quantity 1`] = `
Object {
  "buy_order_type": "bid",
  "buy_price": "",
  "quantity": 1,
  "quantity_type": "count",
  "sell_order_type": "none",
  "sell_price": "",
}
`;

exports[`reducers buy_order should handle update_quantity_type 1`] = `
Object {
  "buy_order_type": "bid",
  "buy_price": "",
  "quantity": "",
  "quantity_type": "quantity_test_type",
  "sell_order_type": "none",
  "sell_price": "",
}
`;

exports[`reducers buy_order should handle update_sell_order_type 1`] = `
Object {
  "buy_order_type": "bid",
  "buy_price": "",
  "quantity": "",
  "quantity_type": "count",
  "sell_order_type": "sell_order_test_type",
  "sell_price": "",
}
`;

exports[`reducers buy_order should handle update_sell_price 1`] = `
Object {
  "buy_order_type": "bid",
  "buy_price": "",
  "quantity": "",
  "quantity_type": "count",
  "sell_order_type": "none",
  "sell_price": 3,
}
`;
